tests/apps/jsonlayerstest.cpp

```diff
@@ -337,6 +337,26 @@ class HotPlugEventCallback : public hwcomposer::DisplayHotPlugEventCallback {
       display->SetPowerMode(power_mode);
   }
 
+  void SetCanvasColor(uint64_t color, uint16_t bpc) {
+    hwcomposer::ScopedSpinLock lock(spin_lock_);
+    PopulateConnectedDisplays();
+
+    if (connected_displays_.empty())
+      return;
+
+    uint64_t uint_max = std::numeric_limits<unsigned int>::max();
+    if (color < 0)
+      color = 0;
+    if (color > uint_max)
+      color = uint_max;
+
+    for (auto &display : connected_displays_)
+      display->SetCanvasColor(bpc, color & 0x000000ff,
```

Comment:
Lines 355 - 357 bit manipulation assume an 8-bit bpc and 32-bit color.  So if bpc happens to be 10 bits, this would be an issue because two bits are getting truncated.  I think you have to handle this as a 64-bit depth instead.

Reply:
Right, the above logic indeed does not handle the 10 bpc case. However, my goal here with this patch is just to demonstrate how the hardware feature works and not to comprehensively cover all the formats. Including cases like 10 bpc, 12 bpc, 16 bpc and testing them would certainly increase the scope of what I planned on doing and I am not sure if I'd be able to do it any soon.

Reply:
In this case can you document the restriction as part of the API in NativeDisplay, so everyone can easily figure this out. If this is something we need to improve in future, you can add a issue in Github issue tracker